qemu/tests/qemu_guest_agent.py

```diff
@@ -4168,15 +4184,50 @@ def _base_on_bg_check_resource_leak():
 
             if bg.is_alive():
                 qga_handles, qga_memory = _process_qga_resource()
-                # Generally the qga handles won't exceed 300, and the memory
-                # setting '3000' is an average value when the guest is not
-                # running redundant operations and apps.
-                if qga_handles > 300 or qga_memory > 3000:
-                    test.fail("QGA commands caused resource leak. handles are"
-                              " %s, memory is %s" % (qga_handles, qga_memory))
-                else:
-                    LOG_JOB.info("Current qga handles is %s, qga memory"
-                                 "is %s", qga_handles, qga_memory)
+                if qga_memory > 3000:
+                    qga_memory_previous = qga_memory
+                    qga_handles_previous = qga_handles
+                    qga_handles_latest, qga_memory_latest = _process_qga_resource()
+                    memory_high_peak = []
+                    memory_low_peak = []
+                    memory_list = []
+                    start_time = time.time()
+                    while int(time.time() - start_time) < 600:
+                        while qga_memory_latest >= qga_memory_previous:
+                            difference = qga_memory_latest - qga_memory_previous
+                            qga_memory_previous = qga_memory_latest
+                            time.sleep(5)
+                            qga_handles_latest, qga_memory_latest = _process_qga_resource()
+                            if qga_memory_latest < qga_memory_previous:
+                                memory_high_peak.append(qga_memory_previous)
+                            if int(time.time() - start_time) > 600:
+                                break
+
+                        while (int(time.time() - start_time) < 600 and
+                               qga_memory_latest <= qga_memory_previous):
+                            difference = qga_memory_latest - qga_memory_previous
+                            qga_memory_previous = qga_memory_latest
+                            time.sleep(5)
+                            qga_handles_latest, qga_memory_latest = _process_qga_resource()
+                            if qga_memory_latest > qga_memory_previous:
+                                memory_low_peak.append(qga_memory_previous)
```

Comment:
Even you named them `low_peak` `high_peak`, these two list only record where is the point of inflection, but only the trend. I suppose you want to append `memory_high_peak` after first `while` break out, and `memory_low_peak` after second `while` break out; and you might want to fail the case if it never enter into second `while`.

Reply:
> Even you named them `low_peak` `high_peak`, these two lists do not record where is the point of inflection, but only the trend. I suppose you want to append `memory_high_peak` after first `while` break out, and `memory_low_peak` after second `while` break out; and you might want to fail the case if it never enter into second `while`.

maybe we can have a meeting for discuss it.